graphrag/index/flows/generate_text_embeddings.py

```diff
@@ -0,0 +1,163 @@
+            if final_entities is not None
+            else None,
+            "column_to_embed": "name",
+            "filename": "create_final_entities_name_embeddings",
+            "base_text_embed": name_text_embed,
+        },
+        entity_description_embedding: {
+            "data": final_entities.loc[:, ["id", "name", "description"]].assign(
+                name_description=lambda df: df["name"] + ":" + df["description"]
+            )
+            if final_entities is not None
+            else None,
+            "column_to_embed": "name_description",
+            "filename": "create_final_entities_description_embeddings",
+            "base_text_embed": name_description_text_embed,
+        },
+        community_title_embedding: {
+            "data": final_community_reports.loc[
+                :, ["id", "full_content", "summary", "title"]
+            ]
+            if final_community_reports is not None
+            else None,
+            "column_to_embed": "title",
+            "filename": "create_final_community_reports_title_embeddings",
+            "base_text_embed": title_text_embed,
+        },
+        community_summary_embedding: {
+            "data": final_community_reports.loc[
+                :, ["id", "full_content", "summary", "title"]
+            ]
+            if final_community_reports is not None
+            else None,
+            "column_to_embed": "summary",
+            "filename": "create_final_community_reports_summary_embeddings",
+            "base_text_embed": summary_text_embed,
+        },
+        community_full_content_embedding: {
+            "data": final_community_reports.loc[
+                :, ["id", "full_content", "summary", "title"]
+            ]
+            if final_community_reports is not None
+            else None,
+            "column_to_embed": "full_content",
+            "filename": "create_final_community_reports_full_content_embeddings",
+            "base_text_embed": full_content_text_embed,
+        },
+    }
+
+    log.info("Creating embeddings")
+    for field in embedded_fields:
+        await _run_and_snapshot_embeddings(
+            callbacks=callbacks,
+            cache=cache,
+            storage=storage,
+            **embedding_param_map[field],
+        )
+
+
+async def _run_and_snapshot_embeddings(
+    data: pd.DataFrame,
+    column_to_embed: str,
+    filename: str,
+    callbacks: VerbCallbacks,
+    cache: PipelineCache,
+    storage: PipelineStorage,
+    base_text_embed: dict | None = None,
+) -> None:
+    """All the steps to generate single embedding."""
+    if base_text_embed:
+        data["embedding"] = await embed_text(
+            data,
+            callbacks,
+            cache,
+            column=column_to_embed,
+            strategy=base_text_embed["strategy"],
+        )
+
+        data = data.loc[:, ["id", "embedding"]]
+
+        if base_text_embed["strategy"]["vector_store"]["snapshot"] is True:
```

Comment:
See above about `embeddings_snapshot` param, which will just be a top-level bool